ring/sampler_gaussian.go

```diff
@@ -137,15 +182,22 @@ func randFloat64(randomBytes []byte) float64 {
 // to use a secure PRNG instead of math/rand.
 func (g *GaussianSampler) normFloat64() (float64, uint64) {
 
+	ptr := g.ptr
+	buff := g.randomBufferN
+	prng := g.prng
+	buffLen := uint64(len(buff))
+
 	for {
 
-		if g.ptr == uint64(len(g.randomBufferN)) {
-			g.prng.Read(g.randomBufferN)
-			g.ptr = 0
+		if ptr == buffLen {
```

Comment:
there is a lot of time this block of code repeated in this function, it could be easier to read if there is a method that simply returns the next 4/8 bytes and calls the prng if the ptr is at the end of the buffer ?